File: apiaryFrontend/src/app/models/corp.ts
export class Corp {
    id!: number;
    framesBlack!: number;
    framesBrown!: number;
    framesWhite!: number;
    framesEmpty!: number;
    hiveId!: number;

    constructor(id: number, framesBlack: number, framesBrown: number, framesWhite: number, framesEmpty: number, hiveId: number) {
        this.id = id;
        this.framesBlack = framesBlack;
        this.framesBrown = framesBrown;
        this.framesWhite = framesWhite;
        this.framesEmpty = framesEmpty;
        this.hiveId = hiveId;
    }
}

File: apiary/src/main/java/com/apiary/corp/Corp.java
package com.apiary.corp;

import com.apiary.hive.Hive;
import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
public class Corp {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private int framesBlack;
    private int framesBrown;
    private int framesWhite;
    private int framesEmpty;

    @ManyToOne
    @JoinColumn(name = "hive_id")
    private Hive hive;

    // Getters and Setters
}
